src/common/darktable.c

```diff
@@ -1581,6 +1591,21 @@ int dt_worker_threads()
   return 1;
 }
 
+size_t dt_get_available_mem()
+{
+  const int level = darktable.dtresources.level;  
+  const size_t total_mem = darktable.dtresources.total_memory;
+  if(level == DT_RESOURCE_LEVEL_UNRESTRICTED) return total_mem;
+  if(level == DT_RESOURCE_LEVEL_TESTING)      return 8192lu * 1024lu * 1024lu;
+  const size_t available = total_mem / 6 * level;
```

Comment:
6 because the levels are all divided in equal chunk. Ok, but what if a new level is added into the enum? I would certainly prefer here using  (DT_RESOURCE_LEVEL_UNRESTRICTED + 1), no?

Reply:
Anyway, see below. Those `DT_RESOURCE_LEVEL_*` should not be present.

Reply:
fully agreed. there are now defined fractions / 1024  for every calculated parameter.